src/python/gudhi/tensorflow/rips_layer.py

```diff
@@ -0,0 +1,75 @@
+import numpy               as np
+import tensorflow          as tf
+from ..rips_complex     import RipsComplex
+
+############################
+# Vietoris-Rips filtration #
+############################
+
+# The parameters of the model are the point coordinates.
+
+def _Rips(DX, max_edge, dimension):
+    # Parameters: DX (distance matrix), 
+    #             max_edge (maximum edge length for Rips filtration), 
+    #             dimension (homology dimension)
+
+    # Compute the persistence pairs with Gudhi
+    rc = RipsComplex(distance_matrix=DX, max_edge_length=max_edge)
+    st = rc.create_simplex_tree(max_dimension=dimension+1)
+    dgm = st.persistence()
+    pairs = st.persistence_pairs()
+
+    # Retrieve vertices v_a and v_b by picking the ones achieving the maximal
+    # distance among all pairwise distances between the simplex vertices
+    indices, pers = [], []
+    for s1, s2 in pairs:
+        if len(s1) == dimension+1 and len(s2) > 0:
+            l1, l2 = np.array(s1), np.array(s2)
+            i1 = [l1[v] for v in np.unravel_index(np.argmax(DX[l1,:][:,l1]),[len(l1), len(l1)])]
+            i2 = [l2[v] for v in np.unravel_index(np.argmax(DX[l2,:][:,l2]),[len(l2), len(l2)])]
+            indices.append(i1)
+            indices.append(i2)
+            pers.append(st.filtration(s2)-st.filtration(s1))
+    
+    # Sort points with distance-to-diagonal
```

Comment:
Is it documented that the output is sorted this way? Is it useful?

Reply:
That is probably from some time ago, when I needed the PDs to be sorted for some reason (that I forgot). It's actually unnecessary so I removed that part.